core/application/impl/crypto_key_reader/crypto_key_reader_util.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/crypto_key_reader/crypto_key_reader_util.hpp"
+
+#include <openssl/pem.h>
+
+#include "application/impl/key_storage_error.hpp"
+#include "crypto/ed25519_types.hpp"
+#include "common/hexutil.hpp"
+
+namespace kagome::application {
+
+  namespace detail {
+    outcome::result<common::Buffer> readEd25519PrivKeyFromPEM(
+        const boost::filesystem::path &file) {
+      FILE *f = fopen(file.string().c_str(), "r");
+      if (!f) {
+        return KeyStorageError::FILE_READ_ERROR;
+      }
+      EVP_PKEY *key = nullptr;
+      EVP_PKEY *result =
```

Comment:
result probably leaks

Reply:
It should point to the same memory as the **key**.

Reply:
ok